### PR TITLE
Username Service Factory Version with usage

### PR DESCRIPTION
**Pull request by sqreek**
@mixth-sense @macbre @wladekb

### CODE REVIEW

File: includes/specials/SpecialBlockList.php [hunk 1]
```diff
@@ -323,10 +328,11 @@ function formatValue( $name, $value ) {
...
 			case 'ipb_by':
 				if ( isset( $row->by_user_name ) ) {
-					$formatted = Linker::userLink( $value, $row->by_user_name );
-					$formatted .= Linker::userToolLinks( $value, $row->by_user_name );
+					$username = $this->usernameService->getUsername( $value, $row->by_user_name );
+					$formatted = Linker::userLink( $value, $username );
+					$formatted .= Linker::userToolLinks( $value, $username );
 				} else {
-					$formatted = htmlspecialchars( $row->ipb_by_text ); // foreign user?
+					$formatted = htmlspecialchars( $this->usernameService->getUsername( $row->ipb_id, $row->ipb_by_text ) ); // foreign user?
```
Comment (WikiaTech) [inline includes/specials/SpecialBlockList.php:335]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
- MAJOR 7 major
- MINOR 11 minor
- INFO 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [SpecialBlockList.php#L126](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L126): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
2. MAJOR [SpecialBlockList.php#L133](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L133): Reduce this "switch/case" number of lines from 13 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
3. MAJOR [SpecialBlockList.php#L251](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L251): The Cyclomatic Complexity of this function "formatValue" is 21 which is greater than 20 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
4. MAJOR [SpecialBlockList.php#L278](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L278): Reduce this "switch/case" number of lines from 22 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
5. MAJOR [SpecialBlockList.php#L283](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L283): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
6. MAJOR [SpecialBlockList.php#L300](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L300): Reduce this "switch/case" number of lines from 29 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
7. MAJOR [SpecialBlockList.php#L343](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L343): Reduce this "switch/case" number of lines from 23 to at most 10, for example by extracting code into function. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1151)
8. MINOR [SpecialBlockList.php#L63](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L63): Define a constant instead of duplicating this literal "Unblock" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
9. MINOR [SpecialBlockList.php#L79](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L79): Define a constant instead of duplicating this literal "options" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
10. MINOR [SpecialBlockList.php#L88](https://github.com/Wikia/app/blob/89935bde3ec51cf54cf863e58d6b878896533090/includes/specials/SpecialBlockList.php#L88): Define a constant instead of duplicating this literal "class" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: lib/Wikia/src/Service/User/Username/MediawikiUsernameService.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	/**
+	 * return username from database
+	 *
+	 * @param $id int
+	 * @param $fallback string
+	 * @return string username
+	 */
+	public function getUsername( $id, $fallback ) {
+
```
Comment (macbre) [inline lib/Wikia/src/Service/User/Username/MediawikiUsernameService.php:14]: Why the empty line? For future use? :)

**Comment by wladekb** [comment]
Could you fill in the PR description?

**Pull request closed without merging**